File: examples/scenario-runner/python/statsig_wrapper.py
import os
from typing import Dict, Any, Optional, Union
import threading
from statsig_python_core import (
    Statsig,
    StatsigUser,
    StatsigOptions,
)
from statsig import (
    statsig,
    StatsigUser as LegacyStatsigUser,
    StatsigOptions as LegacyStatsigOptions,
    StatsigEvent,
    HashingAlgorithm,
)

SCRAPI_URL = "http://scrapi:8000"


class StatsigWrapper:
    is_core: bool = False

    _statsig: Optional[Statsig] = None
    _user: Optional[Union[StatsigUser, Dict[str, Any]]] = None

    @classmethod
    def initialize(cls) -> threading.Event:
        variant = os.environ.get("SDK_VARIANT")

        if variant == "core":
            cls.is_core = True

            options = StatsigOptions(
                specs_url=f"{SCRAPI_URL}/v2/download_config_specs",
                log_event_url=f"{SCRAPI_URL}/v1/log_event",
                output_log_level="debug",
                # disable_user_agent_parsing=True,
                # disable_country_lookup=True,
            )

            cls._statsig = Statsig("secret-PYTHON_CORE", options)
            return cls._statsig.initialize()

        if variant == "legacy":
            cls.is_core = False

            options = LegacyStatsigOptions(api=f"{SCRAPI_URL}/v1")
            statsig.initialize("secret-PYTHON_LEGACY", options)

            event = threading.Event()
            event.set()
            return event

        raise ValueError(f"Invalid SDK variant: {variant}")

    @classmethod
    def set_user(cls, user_data: Dict[str, Any]):
        if cls.is_core:
            cls._user = StatsigUser(user_id=user_data["userID"])
        else:
            cls._user = LegacyStatsigUser(user_id=user_data["userID"])

    @classmethod
    def check_gate(cls, gate_name: str) -> bool:
        if cls.is_core:
            cls._validate_core_user()
            return cls._statsig.check_gate(cls._user, gate_name)

        return statsig.check_gate(cls._user, gate_name)

    @classmethod
    def log_event(cls, event_name: str):
        if cls.is_core:
            cls._validate_core_user()
            cls._statsig.log_event(cls._user, event_name)
        else:
            event = StatsigEvent(
                user=cls._user,
                event_name=event_name,
            )
            statsig.log_event(event)

    @classmethod
    def get_client_initialize_response(cls):
        if cls.is_core:
            cls._validate_core_user()
            return cls._statsig.get_client_initialize_response(cls._user)
        else:
            return statsig.get_client_initialize_response(
                cls._user, hash=HashingAlgorithm.DJB2
            )

    @classmethod
    def _validate_core_user(cls):
        if not isinstance(cls._user, StatsigUser):
            raise ValueError("User not set or not a StatsigUser")

File: examples/scenario-runner/node/statsig-wrapper.ts
import { Statsig, StatsigUser } from '@statsig/statsig-node-core';
import StatsigLegacy, { StatsigUser as StatsigLegacyUser } from 'statsig-node';

const SCRAPI_URL = 'http://scrapi:8000';

export class StatsigWrapper {
  static isCore = false;

  private static statsig: Statsig;
  private static user: StatsigUser | StatsigLegacyUser | null = null;

  static async initialize() {
    const variant = process.env.SDK_VARIANT!;

    if (variant === 'core') {
      this.isCore = true;
      this.statsig = new Statsig('secret-NODE_CORE', {
        specsUrl: `${SCRAPI_URL}/v2/download_config_specs`,
        logEventUrl: `${SCRAPI_URL}/v1/log_event`,
        // disableCountryLookup: true,
      });

      await this.statsig.initialize();
      return;
    }

    if (variant === 'legacy') {
      this.isCore = false;

      await StatsigLegacy.initialize('secret-NODE_LEGACY', {
        api: `${SCRAPI_URL}/v1`,
      });
      return;
    }

    throw new Error(`Invalid SDK variant: ${variant}`);
  }

  static setUser(userData: Record<string, unknown>) {
    if (this.isCore) {
      const user = StatsigUser.withUserID(userData.userID as string);
      this.user = user;
    } else {
      this.user = userData as StatsigLegacyUser;
    }
  }

  static checkGate(gateName: string) {
    if (this.isCore) {
      this.validateCoreUser();

      return this.statsig.checkGate(this.user as StatsigUser, gateName);
    }

    return StatsigLegacy.checkGateSync(
      this.user as StatsigLegacyUser,
      gateName,
    );
  }

  static logEvent(eventName: string) {
    if (this.isCore) {
      this.validateCoreUser();
      this.statsig.logEvent(this.user as StatsigUser, eventName);
    } else {
      StatsigLegacy.logEvent(this.user as StatsigLegacyUser, eventName);
    }
  }

  static getClientInitResponse() {
    if (this.isCore) {
      this.validateCoreUser();
      return this.statsig.getClientInitializeResponse(this.user as StatsigUser);
    }

    return StatsigLegacy.getClientInitializeResponse(
      this.user as StatsigLegacyUser,
    );
  }

  private static validateCoreUser() {
    if (!(this.user instanceof StatsigUser)) {
      throw new Error('User not set or not a StatsigUser');
    }
  }
}
